extensions/wikia/Chat2/ChatRailController.class.php

```diff
@@ -37,8 +37,15 @@ public function executeAnonLoginSuccess() {
 	}
 
 	public function executeGetUsers() {
+		global $wgUser;
+
 		wfProfileIn( __METHOD__ );
-		$this->users = ChatWidget::getUsersInfo();
+		$usersInfo = $wgUser->isLoggedIn() ? ChatWidget::getUsersInfo() : [];
+		$viewedUsersInfo = ChatWidget::getViewedUsersInfo( $usersInfo );
+		$usersCount = count( $usersInfo );
+
+		$this->users = $viewedUsersInfo;
```

Comment:
I think that $this->setVal('users', $viewedUsersInfo) is more preferred way of passing vars to template.
And the same below